sbndqm/Decode/TPC/SBND/CMakeLists.txt

```diff
@@ -1,19 +1,27 @@
-simple_plugin( SBNDTPCDecoder module
-  sbndqm_Decode_Mode
+#add_subdirectory(ChannelMap)
+# this is borrowed from sbndaq-artdaq-core to make the c7 compiler happy
+
+add_definitions(-Wno-nested-anon-types)
+
+cet_build_plugin( SBNDTPCDecoder art::module
+  SOURCE SBNDTPCDecoder_module.cc
+  LIBRARIES
   sbndaq_artdaq_core::sbndaq-artdaq-core_Overlays_SBND
   sbndaq_artdaq_core::sbndaq-artdaq-core_Overlays_SBND_NevisTPC
-  ${LARDATAOBJ}
-  lardataobj_RawData
+  lardataobj::RawData
   artdaq_core::artdaq-core_Utilities
-  ${ART_UTILITIES}
-  ${FHICLCPP}
-  ${MF_MESSAGELOGGER}
-  ${MF_UTILITIES}
-  ${ART_FRAMEWORK_CORE}
-  ${ROOT_BASIC_LIB_LIST}
+  art::Utilities
+  fhiclcpp::fhiclcpp
+  messagefacility::MF_MessageLogger
+  art::Framework_Core
+  ROOT::Core
+  ROOT::Tree 
+  #sbndqm_Decode_TPC_SBND_ChannelMap_TPCChannelMapService_service
+  sbndcode_ChannelMaps_TPC_TPCChannelMapService_service
```

Comment:
`sbndcode_ChannelMaps_TPC_TPCChannelMapService_service` can't be linked at ICARUS